backend/app/component_handler.go

```diff
@@ -0,0 +1,195 @@
+package app
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/codescalersinternships/Flyspray/models"
+	"github.com/mattn/go-sqlite3"
+	"github.com/rs/zerolog/log"
+	"gorm.io/gorm"
+
+	"github.com/gin-gonic/gin"
+)
+
+type createComponentInput struct {
+	Name      string `json:"name" binding:"required"`
+	ProjectID string `json:"project_id" binding:"required"`
+}
+
+type updateComponentInput struct {
+	Name string `json:"name" binding:"required"`
+}
+
+func (a *App) createComponent(ctx *gin.Context) (interface{}, Response) {
+	var input createComponentInput
+
+	if err := ctx.BindJSON(&input); err != nil {
+		log.Error().Err(err).Send()
+		return nil, BadRequest(errors.New("failed to read component data"))
+	}
+
+	_, exists := ctx.Get("user_id")
+	if !exists {
+		return nil, UnAuthorized(errors.New("authentication is required"))
+	}
+
+	newComponent := models.Component{Name: input.Name, ProjectID: input.ProjectID}
```

Comment:
add a check that the user creating the component is the project admin

Reply:
or a member

Reply:
I added check for project admin, should I add that any member can create component, or the one how create the project